### DISCUSSION TITLE
MQTT Issue with Sunton Display

### DISCUSSION THREAD

**Discussion by wilwizard1979** [HASwitchPlate/openHASP#445] [General]
### Perform all steps below and tick them with [x]
- [x ] Check the related part of the [Documentation](https://haswitchplate.github.io/openHASP-docs/)
- [x ] Update openHASP to the latest version
- [x ] Reproduce the issue and describe all steps

### Describe the bug

The Sunton Display is not discovered in the MQTT in Home Assistant. Would that be an issue because the Display is still in Beta Version according to OpenHASP documentation? (Reference: https://www.openhasp.com/0.6.3/devices/sunton-screens/)
The hardware part of the Display is working fine. I have even already added some buttons for testing, and this seems to work fine.
I configured MQTT in the display as picture below
After re-starting the Display, I can see the status of the MQTT communication ok (see picture)
Also in the MQTT Explorer we see it up and running (picture)
The problem is that in my MQTT Integration, nothing else than my current HASS.Agent is discovered

### To Reproduce
The only simple code I have in the Display is below (just for testing)....

```json
{
    "page": 1,
    "id": 10,
    "w": 240,
    "obj": "label",
    "text": "%hostname%"
}

{"comment":" ----------- Buttons ------------"}

{"page":0,"id":1,"obj":"btn","action":"prev","x":0,"y":429,"w":106,"h":50,"bg_color":"#2C3E50","text":"\uE141","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":28}
{"page":0,"id":2,"obj":"btn","action":"back","x":107,"y":429,"w":106,"h":50,"bg_color":"#2C3E50","text":"\uE2DC","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":38}
{"page":0,"id":3,"obj":"btn","action":"next","x":214,"y":429,"w":106,"h":50,"bg_color":"#2C3E50","text":"\uE142","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":28}

{"page":2,"comment":" ---- Simple Circle ----"}
{"page":2,"id":1,"obj":"obj","x":40,"y":100,"w":160,"h":160,"radius":100,"opacity":100,"border_opa":160,"border_width":4,"comment":"touch-catcher"}

{"page":3,"comment":" ---- A Label and a Button with a Text ----"}
{"page":3,"id":1,"obj":"label","x":5,"y":5,"h":50,"w":50,"text":"Hello","enabled":true,"hidden":false}
{"page":3,"id":2,"obj":"btn","x":5,"y":90,"h":90,"w":100,"text":"World","enabled":false,"hidden":false}
```

### Expected behavior
I would expect to see my display listed in the MQTT Integration in Home Assistant

### Screenshots or video

![Img1](https://user-images.githubusercontent.com/108979356/224249228-8895964e-aead-4ce8-865a-828cdc3dbf11.jpeg)

![img2](https://user-images.githubusercontent.com/108979356/224249244-ccd55b73-1f89-4391-ac19-5131051a1dc6.jpeg)

![img3](https://user-images.githubusercontent.com/108979356/224249335-339c408b-df37-45a1-95a9-6f515fd7eb04.jpeg)

![img4](https://user-images.githubusercontent.com/108979356/224249349-61e14d97-8be9-4a86-8369-c70c7e3a90f1.jpeg)

![img5](https://user-images.githubusercontent.com/108979356/224249417-0ee34c3e-efaf-400d-8468-4c6f9c13f39c.jpeg)

Discovery Mode Enabled
![image](https://user-images.githubusercontent.com/108979356/224275747-ac7a43c5-c78e-4c9f-a87f-cdf958b34570.png)

**Comment by fvanroie** [comment]
Did you install the [Home Assistant Community Store (HACS)](https://hacs.xyz/)?
Please install the [openHASP Custom Component](https://www.openhasp.com/0.7.0/integrations/home-assistant/howto/) using HACS.

Make sure to enable `Show Beta versions` and select version `0.7.0` of the openHASP component in HACS.
![image](https://user-images.githubusercontent.com/15969459/224320658-5231276c-5123-444b-8068-59a0124b4d74.png)

**Reply by zalves** [reply]
hello, I've tried to install 0.7.0, but I don't have the option to enable betas when downloading via HACS

**Comment by wilwizard1979** [comment]
Thanks fvanroie, I think I lost something that I had to install this new Custom Component.
I did install OpenHASP custom component at the version shown above (0.7.0) and HACS at version 1.31.0.
Now I have a new problem

After rebooting, the OpenHASP Integration was auto-discovered. 
![image](https://user-images.githubusercontent.com/108979356/224442932-775f6041-17bf-4b83-af17-a0fbb903d1e7.png)

I just kept the default configuration and didn´t select the path for the JSONL file.
![image](https://user-images.githubusercontent.com/108979356/224442982-1d5fcd0b-7f30-46e3-8c6c-a2cc4fcec613.png)

When clicking to submit, I received the following message:
`the integration could not be loaded, try to restart home assistant.`
![image](https://user-images.githubusercontent.com/108979356/224442764-5cd6fa2f-fc49-4f7e-ab92-bad533375098.png)

 
After clicking finish and re-starting Home Assistant, the integration looks like this:
![image](https://user-images.githubusercontent.com/108979356/224441456-2444569f-319d-4ab2-bb5b-1cbea61e22fd.png)

Important to mention that I changed the name of the plate from the first post. Now it is called “plate_sala” which was properly found.

**Comment by wilwizard1979** [comment]
Hi @fvanroie, anything I could check or that I might be doing wrong?

**Comment by kzim200** [comment]
I was seeing exactly the same message.
I was not able to get discovery to work with the openhasp custom integration installed via hacs.
After upgrading the firmware to .7 on the Sunton 1-8048S070C, then removing the openhasp integration from HACS, Rebooting, Reinstalling the Integration V .7, and rebooting again, the screen is discovered but I am not able to interact with in via Home Assistant.
The integration gives me the same error
![openhasp not loaded](https://user-images.githubusercontent.com/20750976/224574108-07bd11fe-a781-44e6-ac67-730a547cd3b5.JPG)

I then went back to the docs:

changed my pages.json file ...
{"page":1,"id":1,"obj":"btn","x":0,"y":0,"w":240,"h":30,"text":"openHASP","value_font":22,"bg_color":"#2C3E50","text_color":"#FFFFFF","radius":0,"border_side":0}
{"page":1,"id":2,"obj":"btn","x":10,"y":40,"w":105,"h":90,"toggle":true,"text":"\uE335","text_font":32,"mode":"break","align":1}
{"page":1,"id":3,"obj":"dropdown","x":10,"y":140,"w":170,"h":30,"options":"Apples\nBananas\nOranges\nMelon"}
{"page":0,"id":1,"obj":"label","x":175,"y":5,"h":30,"w":62,"text":"00.0°C","align":2,"bg_color":"#2C3E50","text_color":"#FFFFFF"}

then updated my configuration file ...
openhasp:
  plate35:
    objects:
      - obj: "p0b1"  # temperature label on all pages
        properties:
          "text": '{{ states("sensor.my_room_temperature") }}°C'
      - obj: "p1b2"  # light-switch toggle button
        properties:
          "val": '{{ 1 if states("light.my_room") == "on" else 0 }}'
          "text": '{{ "\uE6E8" if is_state("light.my_room", "on") else "\uE335" | e }}'
        event:
          "up":
            - service: homeassistant.toggle
              entity_id: "light.my_room"
      - obj: "p1b3"  # dropdown
        event:
          "changed":
            - service: persistent_notification.create
              data:
                message: I like {{ text }}

Rebooted one more time and it is working.

**Reply by wilwizard1979** [reply]
You have no idea how you made my Sunday happier!
It worked also for me. Thanks a lot!!

**Comment by elik745i** [comment]
normally after couple of reboots its discovered, dont know why not instant

**Comment by pauldogg** [comment]
I have a wt32-sc01-plus_16MB that will not become discovered.

openhasp 0.7.0-rc10 fb2bebf

I've tried changing HACS  to 0.7 also 0.7.1, rebooted the device and home assistant with no discovery...

MQTT shows as connected in the information tab of the device.


MQTT
--
Server | XX.XX.XX.XXX
Username | XXXXX
Client ID | plate_36b164
Status | Connected
Received
Published | 32
Failed


Pages.json:

{"page":1,"id":1,"obj":"btn","x":0,"y":0,"w":240,"h":30,"text":"openHASP","value_font":22,"bg_color":"#2C3E50","text_color":"#FFFFFF","radius":0,"border_side":0}
{"page":1,"id":2,"obj":"btn","x":10,"y":40,"w":105,"h":90,"toggle":true,"text":"\uE335","text_font":32,"mode":"break","align":1}
{"page":1,"id":3,"obj":"dropdown","x":10,"y":140,"w":170,"h":30,"options":"Apples\nBananas\nOranges\nMelon"}
{"page":0,"id":1,"obj":"label","x":175,"y":5,"h":30,"w":62,"text":"00.0°C","align":2,"bg_color":"#2C3E50","text_color":"#FFFFFF"}

openhasp.yaml:

  plate35:
    objects:
      - obj: "p0b1"  # temperature label on all pages
        properties:
          "text": '{{ states("sensor.my_room_temperature") }}°C'
      - obj: "p1b2"  # light-switch toggle button
        properties:
          "val": '{{ 1 if states("light.my_room") == "on" else 0 }}'
          "text": '{{ "\uE6E8" if is_state("light.my_room", "on") else "\uE335" | e }}'
        event:
          "up":
            - service: homeassistant.toggle
              entity_id: "light.my_room"
      - obj: "p1b3"  # dropdown
        event:
          "changed":
            - service: persistent_notification.create
              data:
                message: I like {{ text }}

Any help would be greatly appreciated.
Thanks

**Reply by LarryBoy1975** [reply]
I had exactly the same problem!  I compiled a wt32-sc01-plus_16MB and verified it was working with MQTT_Explorer but it wouldn't auto-discover in HomeAssistant.  However, I followed [fvanroie]'s advice from above: "Make sure to enable Show Beta versions and select version 0.7.0 of the openHASP component in HACS" (in my case: 0.7.2). Then i copied the contents of my modified pages.json file (had to modify it because the wt32's screen is much larger than the two lanbon-l8's i already had running) from the "File Editor" on the wt32's configuration webpage to a file i added/re-created on my RPi4 via HomeAssistant's "File editor" (because i could never successfully configure a plate without filling in the "Full path to the JSONL file: custom_components/openhasp/pages.json"), and it finally auto-discovered.  Now all three plates are working in harmony.

**Comment by alfem** [comment]
I am in the very same case than @pauldogg 

My new Sunton 8048S050C does not show up in Home Assistant. I tried to restart HASS, deleting yaml config... but no luck.

I updated the custom component to 0.7.2, and flashed my board using 0.7.0-rc12 cd34d89 build. 

I can see mqttt messages from the board going to Mosquitto, but HASS does not discover it. One hour ago I flashed and connected to HASS a smaller board (esp32-3248s035c_4MB) and worked a treat.

When I enable component debug, these lines show up in HASS log:

```
024-05-28 15:14:45.747 DEBUG (MainThread) [custom_components.openhasp.config_flow] Discovered ZeroConf: {'dim': '[]', 'light': '[]', 'power': '[]', 'input': '{}', 'uri': 'http://192.168.1.123', 'node_t': 'hasp/plate3/', 'sw': '0.7.0-rc12', 'pages': '12', 'hwid': 'f0f5bd4a0568', 'mf': 'openHASP', 'mdl': 'Sunton 1-8048S070C', 'node': 'plate3'}
2024-05-28 15:14:45.748 DEBUG (MainThread) [custom_components.openhasp.config_flow] [hasp/plate3] Discovered pages = 12
2024-05-28 15:14:45.748 DEBUG (MainThread) [custom_components.openhasp.config_flow] [hasp/plate3] Discovered power = []
2024-05-28 15:14:45.748 DEBUG (MainThread) [custom_components.openhasp.config_flow] [hasp/plate3] Discovered light = []
2024-05-28 15:14:45.748 DEBUG (MainThread) [custom_components.openhasp.config_flow] [hasp/plate3] Discovered dim = []
2024-05-28 15:14:45.748 DEBUG (MainThread) [custom_components.openhasp.config_flow] [hasp/plate3] Discovered input = {}

```
and nothing else happens.

Any clue will be welcomed.

**Comment by bh213** [comment]
@alfem I had the same issue, got it working by commenting out `

```
self._abort_if_unique_id_configured()
``` 
in `config_flow.py`. 

My place hardware id got stored somewhere and then it would always abort in this line. Hope this helps.

**Reply by alfem** [reply]
Yes!! I commented out that line in the custom component, restarted hass and my new plate quickly showed up in lovelace.

Thanks a lot for the workaround, Gorazd

**Reply by fvanroie** [reply]
Plea se post your issue and solution in https://github.com/HASwitchPlate/openHASP-custom-component/ for a permanent fix.

**Reply by alfem** [reply]
> Plea se post your issue and solution in https://github.com/HASwitchPlate/openHASP-custom-component/ for a permanent fix.

[Done!](https://github.com/HASwitchPlate/openHASP-custom-component/issues/128)

**Comment by LarryBoy1975** [comment]
I posted about having this issue previously above while trying to add a wt32-sc01-plus_16MB with two existing Lanbon-l8's and found success by following [fvanroie]'s advice. I like the pre-built form factor of the wt32 so much (i can directly mount it to a single or multi light-switch plate by cutting a hole in it with a Dremel) that i got a second one and then encountered the same issue again; however, this time i already had one wt32 running and it just wouldn't auto-discover the second one. I followed [bh213]'s suggestion and commented out "self._abort_if_unique_id_configured() in config_flow.py" and now the second wt32 is working but, even though they are both configured, HA keeps re-discovering both plates and keeps asking to configure them as though they are new.  I'm sure all i have to do is undo the comment-out, reboot, and it should stop asking, but obviously some discrimination improvements in the discovery routine are in order.

**Reply by fvanroie** [reply]
This needs to be posted to the custom-component repository, as this is very off-topic for this thread.
